flutter/lib/src/integrations/flutter_error_integration.dart

```diff
@@ -59,7 +59,8 @@ class FlutterErrorIntegration extends Integration<SentryFlutterOptions> {
         // FlutterError doesn't crash the App.
         final mechanism = Mechanism(
           type: 'FlutterError',
-          handled: true,
+          handled: _defaultOnError != null &&
+              _defaultOnError != FlutterError.presentError,
```

Comment:
```suggestion
          handled: false,
```

Not sure why you did this conditional if, mind explaining?

Reply:
Thought that users could provide their own callback to `FlutterError.onError`, which we respect and also call, so I assumed that in this case, we'd mark it as handled.

Reply:
If they provided `FlutterError.onError`, they most likely have overridden ours, so I'd still keep the same behavior.